src/modules/acbs/acbs-party.service.get-party-by-search-text.test.ts

```diff
@@ -0,0 +1,144 @@
+import { HttpService } from '@nestjs/axios';
+import { GetPartyGenerator } from '@ukef-test/support/generator/get-party-generator';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import { AxiosError } from 'axios';
+import { when } from 'jest-when';
+import { of, throwError } from 'rxjs';
+
+import { DateStringTransformations } from '../date/date-string.transformations';
+import { AcbsPartyService } from './acbs-party.service';
+import { AcbsException } from './exception/acbs.exception';
+
+describe('AcbsPartyService', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const dateStringTransformations = new DateStringTransformations();
+  const idToken = valueGenerator.string();
+  const baseUrl = valueGenerator.httpsUrl();
+
+  let httpService: HttpService;
+  let service: AcbsPartyService;
+
+  let httpServiceGet: jest.Mock;
+
+  beforeEach(() => {
+    httpService = new HttpService();
+
+    httpServiceGet = jest.fn();
+    httpService.get = httpServiceGet;
+
+    service = new AcbsPartyService({ baseUrl }, httpService);
+  });
+
+  describe('getPartyBySearchText', () => {
+    const searchText = 'searchText';
+
+    it('returns the party if ACBS responds with the party', async () => {
+      const { partiesInAcbs } = new GetPartyGenerator(valueGenerator, dateStringTransformations).generate({ numberToGenerate: 1 });
+      const partyInAcbs = partiesInAcbs[0];
```

Comment:
Do we need to add index `0` void check?

Reply:
`partiesInAcbs` is test data here so we know that index `0` is always going to exist

Reply:
@francescastocco can we change it to
```
const [partyInAcbs] = partiesInAcbs;
```